staging/src/k8s.io/apiserver/pkg/util/wsstream/conn.go

```diff
@@ -265,6 +267,18 @@ func (conn *Conn) handle(ws *websocket.Conn) {
 			klog.V(6).Infof("Frame is targeted for a reader %d that is not valid, possible protocol error", channel)
 			continue
 		}
+		// Half-close the stream when receiving a frame with the STDIN channel id, but
+		// with empty message data. This frame is interpreted as a Close
+		// signal and closes STDIN stream on this server-side. This protocol allows
+		// the command to complete and the remaining output to be sent to the client
+		// before the WebSockets Connection is closed.
+		//
+		// See https://github.com/kubernetes/kubernetes/issues/89899#issuecomment-1132502190.
+		if int(channel) == stdinChannelId && len(data) == 0 {
+			klog.V(4).Infof("Received half-close signal from client; close STDIN stream")
+			conn.channels[int(channel)].Close()
+			continue // MUST continue and call Receive again.
```

Comment:
Why we must continue and call `Receive` again?. Because Receive gets `Error on socket receive: read tcp 127.0.0.1:35867->127.0.0.1:52940: use of closed network connection` error instead EOF. I think somewhere closes this connection prematurely.